### PR TITLE
Match Mirage config to API

### LINKED ISSUE #495: Go through mirage config and ensure it matches our API completely
# Problem

We need to ensure our mirage server config is actually mirroring our API. A thorough pass through it should do the trick, but it would be a good idea to circle back every now and then to.

* https://github.com/code-corps/code-corps-api is our API repo.
* https://github.com/code-corps/code-corps-api/blob/develop/web/router.ex would be the first step
* We should also look at the controllers and tests to see if there's any special behavior

## Also

* Mirage models should match up app models
* Both mirage and app models should match up API models and views. Views are probably more important here, especially with how relationships are defined

### PR DESCRIPTION
**Pull request by JoshSmith**
# What's in this PR?

- Added missing endpoints
- Added missing coalesces

## References
Fixes #495 

Also identified that the API has two endpoints that are not in Mirage, and may not be necessary to have in the API:

- https://github.com/code-corps/code-corps-api/issues/340
- https://github.com/code-corps/code-corps-api/issues/341

### CODE REVIEW

File: mirage/config.js [hunk 1]
```diff
@@ -469,21 +535,30 @@ export default function() {
...
   this.delete('/user-roles/:id');
 
   //////////////
   // User skills
   //////////////
 
-  // GET /user-skills
-  this.get('/user-skills');
+  // POST /user-skills
+  this.post('/user-skills', { coalesce: true });
```
Comment (JoshSmith) [inline mirage/config.js]: This needs fixed.

File: mirage/config.js [hunk 2]
```diff
@@ -469,21 +535,30 @@ export default function() {
   /////////////
 
   // POST /user-roles
+  this.post('/user-roles', { coalesce: true });
```
Comment (JoshSmith) [inline mirage/config.js]: This needs fixed.

Comment (JoshSmith) [inline mirage/config.js]: Should be a `GET`.

File: mirage/config.js (hunk 1 above)
Comment (JoshSmith) [inline mirage/config.js]: Should be a `GET`.

**Comment by JoshSmith** [comment]
Fixed, needs review. @begedin?

File: mirage/config.js [hunk 3]
```diff
@@ -178,26 +198,41 @@ export default function() {
...
+
   ///////////
   // Projects
   ///////////
 
   // GET /projects
+  this.get('/projects', { coalesce: true });
+
+  // POST /projects
   this.get('/projects');
```
Comment (begedin) [inline mirage/config.js]: Should be a post

**Pull request merged**